hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/impl/connector/hadoop/ReadHdfsP.java

```diff
@@ -83,7 +83,12 @@ private ReadHdfsP(@Nonnull List<RecordReader> recordReaders, @Nonnull Distribute
 
     @Override
     public boolean complete() {
-        return emitFromTraverser(trav);
+        R nextItem = trav.next();
```

Comment:
we should consider going back to non-blocking outbox here

Reply:
I will also add that we will be yielding with every time, causing additional overhead inside the Tasklet.